### DISCUSSION TITLE
Generate css with a specified filename in next js

### DISCUSSION THREAD

**Discussion by dMaisnam** [tailwindlabs/tailwindcss#7517] [Help]
Hello. So working on my nextjs (version 12) project, I came across this situation where I need to export the files as static html files. Whenever I ran `next build && next export`, the tailwind css file generated in `out/_next/static/css` directory always get as a `<random -string>.css` filename. So I had to manually edit all the individual html files generated to link to the stylesheet . 

Is there a way how we can generate the css file as a named file so that I can include the css path to the `_document` file and don't have to manually edit them as I have done?

Any help is much appreciated. Thank you for your time.

**Comment by chailandau** [comment]
Hi from the future -- I ran into this same issue. Not sure if you ever solved it, but I found using the Tailwind CLI separate from Next's build process was a good solution.

`npx tailwindcss -i ./src/styles/main.css -o ./dist/styles.css --minify`

^ is assuming you have the Tailwind CLI installed already, which you should if its a part of your Next build. Here are the [CLI docs](https://tailwindcss.com/docs/installation), and obviously, change paths as needed.
